### PR TITLE
[Enhancement] User friendly error message when AWS parseRegion gets error

### PR DESCRIPTION
**Pull request by zhenxiao**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [X] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [X] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [X] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/fs/hdfs/HdfsFsManager.java [hunk 1]
```diff
@@ -1216,6 +1216,14 @@ public List<TBrokerFileStatus> listPath(String path, boolean fileNameOnly, Map<S
         } catch (FileNotFoundException e) {
             LOG.info("file not found: " + path, e);
             throw new UserException("file not found: " + path, e);
+        } catch (IllegalArgumentException e) {
+            if (e.getMessage().equals("hostname cannot be null")) {
+                LOG.error("Error when parsing region", e);
+                throw new UserException("Error when parsing region", e);
+            } else {
+                LOG.error("errors while get file status ", e);
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/fs/hdfs/HdfsFsManager.java:1224]:
```
LOG.error("The arguments of blob store(S3/Azure) may be wrong. You can check the arguments like region, IAM, instance profile and so on.");
```

File: fs_brokers/apache_hdfs_broker/src/main/java/com/starrocks/broker/hdfs/FileSystemManager.java [hunk 2]
```diff
@@ -905,6 +905,16 @@ public List<TBrokerFileStatus> listPath(String path, boolean fileNameOnly, Map<S
             logger.info("file not found: " + e.getMessage());
             throw new BrokerException(TBrokerOperationStatusCode.FILE_NOT_FOUND,
                     e, "file not found");
+        } catch (IllegalArgumentException e) {
```
Comment (chaoyli) [inline fs_brokers/apache_hdfs_broker/src/main/java/com/starrocks/broker/hdfs/FileSystemManager.java:908]:
```
LOG.error("The arguments of blob store(S3/Azure) may be wrong. You can check the arguments like region, IAM, instance profile and so on.");
```

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 9 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsManager.java|0|9|00.00%|[1219, 1220, 1221, 1222, 1224, 1225, 1227, 1228, 1229]|

**Review by chaoyli — DISMISSED** [review]

**Review by wyb — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 3 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsManager.java|0|3|00.00%|[1219, 1220, 1222]|

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 3 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsManager.java|0|3|00.00%|[1219, 1220, 1222]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 3 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsManager.java|0|3|00.00%|[1219, 1220, 1222]|

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**